src/dsp_tools/analyse_xml_data/xml_to_graph_models.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import annotations
+
+import uuid
+from dataclasses import dataclass
+from typing import Optional
+from uuid import UUID
+
+
+@dataclass
+class TripleGraph:
+    """
+    Contains all classes that express links between resources.
+    """
+
+    resptr_links: Optional[list[ResptrLink]]
+    xml_links: Optional[list[XMLLink]]
+
+
+@dataclass
+class ResptrLink:
+    """
+    This class represents a link between two resources.
+    """
+
+    subject_id: str
+    object_id: str
+    edge_weight: float | int = 1
```

Comment:
could you use float everywhere to avoid union types? it will still work to pass integers, I think.